### PR TITLE
Reorganize

### LINKED ISSUE #12: Provide option to split by location for validation and testing
Splitting by image can lead to data leakage and will give an inflated estimate of how well a model would perform when applied to a new location.

### LINKED ISSUE #27: Experiment with decreasing learning rates as you go further back into the network when fine-tuning
See #2 in https://blog.floydhub.com/ten-techniques-from-fast-ai/

### LINKED ISSUE #30: Support experimenting with learning rate
E.g. try the approach described in Section 3 of https://blog.floydhub.com/ten-techniques-from-fast-ai/

### LINKED ISSUE #32: Support restarts in SGD
See #5 in https://blog.floydhub.com/ten-techniques-from-fast-ai/

### LINKED ISSUE #33: Support image preprocessing (e.g. histogram normalization)

### LINKED ISSUE #43: Build up image sizes
See Section 9 of https://blog.floydhub.com/ten-techniques-from-fast-ai/

### LINKED ISSUE #46: Try fastai library

### LINKED ISSUE #48: Support multitask learning
Train many binary classifiers with shared weights rather than one multiclass classifier.

### LINKED ISSUE #51: Data not available
I am working to get a data set released for use with this code.

### LINKED ISSUE #73: Error in preprocess_images.py if exif data for DateTime  is not present for images
We should add a condition check to stop program from halting even if exif data for DateTime is not present in the images

### PR DESCRIPTION
**Pull request by gsganden**
Closes #73 
Closes #51 
Closes #48 
Closes #46 
Closes #43 
Closes #32 
Closes #33 
Closes #30 
Closes #27 
Closes #12

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,8 @@
-
 # Stuff shared by all builds
 install:
   - python setup.py install
 script:
-  - python -m pytest
+  - :
```
[the remark this answers is not in the document]
Comment (gsganden) [reply .travis.yml]: It does nothing. `python -m pytest` raises an error because there are no tests currently, and we aren't ready to run automated linting, so right now I think it should just make sure that the installation works.

File: setup.py [hunk 2]
```diff
@@ -1,40 +1,30 @@
...
+    long_description=readme,
+    packages=find_packages(exclude=("tests")),
     install_requires=regular_packages,
-    extras_require={
-        'testing': testing_packages + regular_packages
-    }
+    extras_require={"dev": regular_packages + dev_packages},
+    author="Greg Gandenberger",
+    author_email="gsganden@gmail.com",
+    url="https://github.com/ShopRunner/autofocus",
```
[the remark this answers is not in the document]
Comment (gsganden) [reply setup.py]: This change was not intentional, and I don't know how it happened. I am not planning to fork to ShopRunner.

File: setup.py [hunk 3]
```diff
@@ -1,40 +1,30 @@
...
-]
+with open(README_FILE_PATH) as r:
+    readme = r.read()
 
-setuptools.setup(
-    name='autofocus',
-    version='0.2.0',
-    url='https://github.com/UptakeOpenSource/autofocus',
-    python_requires='>=3.6.0',
+regular_packages = ["boto3", "creevey", "opencv-contrib-python", "pandas", "tqdm"]
```
[the remark this answers is not in the document]
Comment (gsganden) [reply setup.py]: I really, really like Creevey.

File: README.md [hunk 4]
```diff
@@ -2,126 +2,50 @@
...
+Install the package in editable model with dev dependencies:
 
-## Steps for Training a Camera Traps Model
-
-### 0. download_images.py
-
-#### Example call
 ```bash
-python autofocus/download_images.py --local-folder data --bucket autofocus
+pip install -e ".[dev]"
```
[the remark this answers is not in the document]
Comment (gsganden) [reply README.md]: It's mostly scripts with a few imports from configs and stuff, so I don't think that approach is appropriate for autofocus. But Creevey is on PyPI!

File: CHANGELOG [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# Changelog
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+
+# [1.0.0] - 2019-4-13
+### Changed
+- Rewrote dataset building code, offloading a lot of complexity to Creevey.
```
[the remark this answers is not in the document]
Comment (gsganden) [reply CHANGELOG]: > call this NEWS.md not CHANGELOG (your call)

We have a convention of using CHANGELOG at ShopRunner (actually CHANGELOG.md, I should change that if nothing else), so I'm inclined to stick with that unless there's a good reason not to. Do you think NEWS.md is a lot more common?

> Add PR and/or issue links after each bullet point so people who want to go deeper can see the exact change

Cool

[the remark this answers is not in the document]
Comment (jameslamb) [reply CHANGELOG]: I'm indifferent, I guess I've seen both. I would at least change to `CHANGELOG.md` as you mentioned, so GitHub will use markdown rendering.

File: setup.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jameslamb) [reply setup.py]: Ok awesome. Then yeah please change back to `UptakeOpenSource`.

I also should tell you...we were finally able to get `github.com/Uptake` so in the next week or two we'll change to `github.com/Uptake`. I'll make a PR to fix this link when I do that, but also GitHub gives you automatic redirects after a change name so nothing should break.

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (jameslamb) [reply README.md]: well one nice thing it lets people do is peg to particular versions in their own applications, without needing to know your versioning conventions. So like imagine some researcher at the San Diego Zoo wants to prototype an app that uses the code in this project, is would be valuable for that person to always be able to `pip install autofocus==2.0.1` so they don't have to keep up with changes on `master`.

Anyway, not a blocker for this PR!

File: .travis.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jameslamb) [reply .travis.yml]: ah I see. I actually disagree with doing this. It means that the first person coming to write tests (if that person isn't you) needs to know to come back and add `python -m pytest`.

I think it would be better to actually put in `python -m pytest` and then just make one test that has `assert True == True` and create one or more "good first issue" issues describing tests you want people to add one day. That way, more junior developers who want to help could focus on writing tests and not have to know about Travis!

[the remark this answers is not in the document]
Comment (gsganden) [reply .travis.yml]: Good idea

File: CHANGELOG (hunk 5 above)
[the remark this answers is not in the document]
Comment (gsganden) [reply CHANGELOG]: Done

File: README.md [hunk 6]
```diff
@@ -2,126 +2,50 @@
...
-
-### 1. preprocess_images.py
-
-#### Example call
+Download a preprocessed version of our primary dataset to `autofocus/data/lpz_2016_2017/processed`:
 
 ```bash
-python autofocus/preprocess_images.py \
---indir data/lpz_data/images_2016 --outdir results/preprocessed_images
+python autofocus/build_dataset/lpz_2016_2017/get_processed.py
```
[the remark this answers is not in the document]
Comment (gsganden) [reply README.md]: See the updates. I replaced the Python scripts with bash commands and used the formatting you suggest.

File: setup.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (gsganden) [reply setup.py]: Fixed

**Comment by gsganden** [comment]
Closes #66
